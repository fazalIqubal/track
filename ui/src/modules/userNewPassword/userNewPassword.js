import React, { Component } from 'react';
import { connect } from 'react-redux';
import { history } from '../../helpers';
import './userNewPassword.scss'
import logo from '../../image/logo1.png'
import { Form, Input, Button, Checkbox } from 'antd'
import { UserOutlined, LockOutlined } from '@ant-design/icons';
import { setUserNewPassword } from '../user/action/user.actions'
import { Toast } from '../../wrapperComponents/notificationToast/notificationToast'
class userNewPassword extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			userPassword: {
				id: ''
			},
			error: ""
		}
	}

	handleChange = (e) => {
		const userPassword = Object.assign({}, this.state.userPassword)
		const name = e.target.name;
		const value = e.target.value;
		userPassword[name] = value;
		this.setState({ userPassword: userPassword });
	}

	getlocalData(userPassword) {
		var localData = JSON.parse(localStorage.getItem('user'))
		if (localData) {
			userPassword.id = localData.user_id
		}
	}

	localClear() {
		localStorage.clear();
		localStorage.removeItem('user');
	}

	handleSubmit = (e) => {

		e.preventDefault();
		const { userPassword } = this.state;
		const { dispatch } = this.props;
		this.getlocalData(userPassword)
		if (userPassword && userPassword.temp_password && userPassword.user_newpassword && (userPassword.user_newpassword.length >= 5) && userPassword.user_confirm_password) {
			if (userPassword.user_newpassword === userPassword.user_confirm_password) {
				dispatch(setUserNewPassword(userPassword))
					.then((res) => {
						if (res && res.statusCode == 201) {
							this.localClear()
						}
						history.push('/login')
					})
			}
		}
	}

	render() {
		const { userPassword } = this.state;
		return (
			<div className='login-container'>
				<div className='app-logo'>
					<img src={logo} />
				</div>
				<div className="passwordPage">
					<h6 className='login-text'>Please enter a new password for your account</h6>
					<Form
						onSubmitCapture={(e) => this.handleSubmit(e)}
					>
						<Form.Item
							name="temp_password"
							rules={[
								{
									required: true,
									message: 'Please input your temp password!',
								},
							]}
						>
							<Input
								prefix={<LockOutlined className="site-form-item-icon" />}
								name="temp_password"
								type="password"
								placeholder="Old Password"
								onChange={(e) => this.handleChange(e)}
								id='temp_password'
								value={userPassword.temp_password}
							/>
						</Form.Item>
						<Form.Item
							name="user_newpassword"
							rules={[
								{
									required: true,
									message: 'Please input your New Password!',
								},
								{ min: 5, message: 'Password must be minimum 5 characters.' },
							]}
							id='passErr'
						>
							<Input
								prefix={<LockOutlined className="site-form-item-icon" />}
								name="user_newpassword"
								type="password"
								placeholder="New Password"
								onChange={(e) => this.handleChange(e)}
								id='user_newpassword'
								value={userPassword.user_newpassword}
							/>
						</Form.Item>
						<Form.Item
							name="user_confirm_password"
							rules={[
								{
									required: true,
									message: 'Please input your Confirm Password!',
								},
								({ getFieldValue }) => ({
									validator(rule, value) {
										if (!value || getFieldValue('user_newpassword') === value) {
											return Promise.resolve();
										}
										return Promise.reject("Password doesn't match!");
									},
								})
							]}
							id='passconfirmEr'
						>
							<Input
								prefix={<LockOutlined className="site-form-item-icon" />}
								name="user_confirm_password"
								type="password"
								placeholder="Confirm Password"
								onChange={(e) => this.handleChange(e)}
								id='user_confirm_password'
								value={userPassword.user_confirm_password}
							/>
						</Form.Item>
						<Form.Item className='singIn-btn'>
							<Button type="primary" htmlType="submit" className="login-form-button" id='submit'>
								Submit
						</Button>
						</Form.Item>
					</Form>
				</div>
			</div>
		)
	}
}

function mapStateToProps(state) {
	const { user } = state.authentication;
	return {
		user,
	}
}
export default connect(mapStateToProps)(userNewPassword);