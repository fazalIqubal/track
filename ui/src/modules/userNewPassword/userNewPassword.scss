
.login-container{
	background-color: #f2f6fa;
	position: absolute;
	width: 100%;
	height: auto;
	min-height: 100vh;
	.passwordPage{
    width: 30%;
    margin: auto !important;
    padding: 40px 45px !important;
    background-color: #fff;
    box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 10px;
    .login-text {
      text-align: center;
      font-weight: 600;
      color: #8b8a8a;
      margin-bottom: 25px;
      font-size: 15px;
    }
    .login-form-button{
      width: 100%;
      background-color: #41b3a3;
      color: white;
      font-size: 11px;
      font-family: nunito-black;
      padding: 0px 65px;
      cursor: pointer; 
      height: 40px;
      border-radius: 8px;
      border: 0;
      font-size: 17px;
      font-weight: 500;
      margin-top: 30px;
    }
    input {
      height: 35px;
      border-radius: 5px;
    }
    span{
      border-radius: 8px;
    }
  }
  
  .login-text {
    text-align: center;
    font-weight: 600;
    color: #8b8a8a;
    margin-bottom: 25px;
    font-size: 15px;
  }

  .mg-top{
		margin-top: 30px;
	}

	.mg-bottom{
		margin-bottom: 30px;
	}
  .errorText{
      color: #ff4d4f;
    }
    .ant-form-item-has-error .ant-input-affix-wrapper, .ant-form-item-has-error .ant-input:hover, .ant-form-item-has-error .ant-input-affix-wrapper:hover {
      border-color: #ff4d4f;
    }
}
