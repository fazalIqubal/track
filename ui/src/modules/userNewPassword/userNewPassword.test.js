import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, mount } from 'enzyme';
import Enzyme from 'enzyme';
import { shallowToJson, toJson } from 'enzyme-to-json';
import userNewPassword from './userNewPassword';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer";
Enzyme.configure({ adapter: new Adapter() })
import logoImage from '../../image/logo1.png'

describe("user new Password test Description", () => {
    let userNewPasswordcomponent;
    const mockDispatch = jest.fn();
    beforeEach(() => {
        userNewPasswordcomponent = shallow(<userNewPassword.WrappedComponent data={[]} dispatch={mockDispatch} />)
    })
    afterEach(() => {
        userNewPasswordcomponent = null;
    })

    it('userNewPassword should render correctly', () => {
        expect(shallowToJson(userNewPasswordcomponent)).toMatchSnapshot();
    });
    it('check the form heading text', () => {
        expect(userNewPasswordcomponent.find('.login-text').text()).toBe('Please enter a new password for your account')
    })
    it('should contain logo', () => {
        expect(userNewPasswordcomponent.find(".app-logo").find('img').prop("src")).toEqual(logoImage);
    })
    it("To check temp password field existence ", () => {
        expect(userNewPasswordcomponent.find('#temp_password').exists()).toEqual(true)
    })
    it('To check user_newpassword field existence ', () => {
        expect(userNewPasswordcomponent.find('#user_newpassword').exists()).toEqual(true)
    })
    it('To check user_confirm_password field existence ', () => {
        expect(userNewPasswordcomponent.find('#user_confirm_password').exists()).toEqual(true)
    })
    it('To check submit Button existence ', () => {
        expect(userNewPasswordcomponent.find('#submit').exists()).toEqual(true)
    })
    it('Onchange Temp password Field', () => {
        userNewPasswordcomponent.find('#temp_password').simulate('change', { target: { value: 'temp@123', name: 'temp_password' } });
        expect(userNewPasswordcomponent.find('#temp_password').props().value).toBe('temp@123');
    })
    it('Onchange New password Field', () => {
        userNewPasswordcomponent.find('#user_newpassword').simulate('change', { target: { value: 'Password@123', name: 'user_newpassword' } });
        expect(userNewPasswordcomponent.find('#user_newpassword').props().value).toBe('Password@123');
    })
    it('Onchange confirm password Field', () => {
        userNewPasswordcomponent.find('#user_confirm_password').find({ name: 'user_confirm_password' }).simulate('change', { target: { value: 'comPassword@123', name: 'user_confirm_password' } });
        expect(userNewPasswordcomponent.find('#user_confirm_password').find({ name: 'user_confirm_password' }).props().value).toBe('comPassword@123');
    })
    it('Onchange confirm password Field match with new Password field', () => {
        userNewPasswordcomponent.find('#user_newpassword').simulate('change', { target: { value: 'matchPassword@123', name: 'user_newpassword' } });
        var matchpassword = userNewPasswordcomponent.find('#user_newpassword').props().value
        userNewPasswordcomponent.find('#user_confirm_password').simulate('change', { target: { value: 'matchPassword@123', name: 'user_confirm_password' } });
        expect(userNewPasswordcomponent.find('#user_confirm_password').props().value).toEqual(matchpassword);
    })

    it("error message for user temp password", () => {
        expect(userNewPasswordcomponent.find({ name: 'temp_password' }).first().props().rules[0].message).toBe('Please input your temp password!')
    })
    it("error message for user New password", () => {
        expect(userNewPasswordcomponent.find({ name: 'user_newpassword' }).first().props().rules[0].message).toBe('Please input your New Password!')
    })
    it("error message for user Confirm password", () => {
        expect(userNewPasswordcomponent.find({ name: 'user_confirm_password' }).first().props().rules[0].message).toBe('Please input your Confirm Password!')
    })
    it('submit button text check', () => {
        expect(userNewPasswordcomponent.find('#submit').text()).toEqual('Submit')
    })

})


