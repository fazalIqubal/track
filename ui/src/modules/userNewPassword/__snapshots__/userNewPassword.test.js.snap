// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user new Password test Description userNewPassword should render correctly 1`] = `
<div
  className="login-container"
>
  <div
    className="app-logo"
  >
    <img
      src="logo1.png"
    />
  </div>
  <div
    className="passwordPage"
  >
    <h6
      className="login-text"
    >
      Please enter a new password for your account
    </h6>
    <ForwardRef(InternalForm)
      onSubmitCapture={[Function]}
    >
      <FormItem
        name="temp_password"
        rules={
          Array [
            Object {
              "message": "Please input your temp password!",
              "required": true,
            },
          ]
        }
      >
        <Input
          id="temp_password"
          name="temp_password"
          onChange={[Function]}
          placeholder="Old Password"
          prefix={
            <ForwardRef(LockOutlined)
              className="site-form-item-icon"
            />
          }
          type="password"
        />
      </FormItem>
      <FormItem
        id="passErr"
        name="user_newpassword"
        rules={
          Array [
            Object {
              "message": "Please input your New Password!",
              "required": true,
            },
            Object {
              "message": "Password must be minimum 5 characters.",
              "min": 5,
            },
          ]
        }
      >
        <Input
          id="user_newpassword"
          name="user_newpassword"
          onChange={[Function]}
          placeholder="New Password"
          prefix={
            <ForwardRef(LockOutlined)
              className="site-form-item-icon"
            />
          }
          type="password"
        />
      </FormItem>
      <FormItem
        id="passconfirmEr"
        name="user_confirm_password"
        rules={
          Array [
            Object {
              "message": "Please input your Confirm Password!",
              "required": true,
            },
            [Function],
          ]
        }
      >
        <Input
          id="user_confirm_password"
          name="user_confirm_password"
          onChange={[Function]}
          placeholder="Confirm Password"
          prefix={
            <ForwardRef(LockOutlined)
              className="site-form-item-icon"
            />
          }
          type="password"
        />
      </FormItem>
      <FormItem
        className="singIn-btn"
      >
        <Button
          block={false}
          className="login-form-button"
          ghost={false}
          htmlType="submit"
          id="submit"
          loading={false}
          type="primary"
        >
          Submit
        </Button>
      </FormItem>
    </ForwardRef(InternalForm)>
  </div>
</div>
`;
